### PR TITLE
VP-6011: Implement an ability to precise tune AppInsights telemetry

### PR DESCRIPTION
**Pull request by akak1977**

### CODE REVIEW

**Comment by akak1977** [comment]
These changes checked on Azure:
Telemetry processors ordered as desired, and the settings applied.
image

File: src/VirtoCommerce.Platform.Web/Telemetry/ApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            var samplingOptions = app.ApplicationServices.GetRequiredService<IOptions<SamplingOptions>>().Value;
+
+            var configuration = app.ApplicationServices.GetService<TelemetryConfiguration>();
+
+            var builder = configuration.DefaultTelemetrySink.TelemetryProcessorChainBuilder;
+            if (samplingOptions.Processor == SamplingProcessor.Adaptive)
+            {
+                // Using adaptive rate sampling
+                builder.Use(x =>
+                    new AdaptiveSamplingTelemetryProcessor(samplingOptions.Adaptive, null, x)
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Web/Telemetry/ApplicationBuilderExtensions.cs:32]: Does it seem that not all SamplingOptions properties are used here?

```
"InitialSamplingPercentage": "100",
                    "MinSamplingPercentage": "0.1",
                    "MaxSamplingPercentage": "100",
                    "EvaluationInterval": "00:00:15",
                    "SamplingPercentageDecreaseTimeout": "00:02:00",
                    "SamplingPercentageIncreaseTimeout": "00:15:00",
```

File: src/VirtoCommerce.Platform.Core/Telemetry/SamplingOptions.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+using Microsoft.ApplicationInsights.WindowsServer.Channel.Implementation;
+
+namespace VirtoCommerce.Platform.Core.Telemetry
+{
+    /// <summary>
+    /// AppInsights sampling options
+    /// </summary>
+    public class SamplingOptions
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Core/Telemetry/SamplingOptions.cs:8]: I think better would be to use a single POCO type for all AI settings, and combine everything into one config. Also include **EnableSqlCommandTextInstrumentation** and **IgnoreSqlTelemetryOptions** settings into one options type.

File: src/VirtoCommerce.Platform.Web/Telemetry/ApplicationBuilderExtensions.cs (hunk 1 above)
Comment (akak1977) [reply src/VirtoCommerce.Platform.Web/Telemetry/ApplicationBuilderExtensions.cs:32]: Used all the options for adaptive.

**Review by tatarincev — APPROVED** [review]

**Pull request merged**